### ISSUE TITLE
vtk pin to expat 2.5 compatibility problems

### ISSUE THREAD

**Issue by traversaro** [conda-forge/vtk-feedstock#329]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

The problem was discussed in https://github.com/conda-forge/vtk-feedstock/pull/321, and it was solved by constraining libexpat to be <= 2.6 . However, as this was expected this is starting to create problem, for example I just experienced a failure of this kind:

~~~
C:\Users\straversaro>conda create -n test24 vtk==9.2.6 opencv==4.10.0 gz-sim8==8.1.0
Channels:
 - robostack-staging
 - conda-forge
Platform: win-64
Collecting package metadata (repodata.json): done
Solving environment: failed

LibMambaUnsatisfiableError: Encountered problems while solving:
  - package opencv-4.10.0-qt5_py310h5979ca7_500 requires libprotobuf >=4.25.3,<4.25.4.0a0, but none of the providers can be installed

Could not solve for environment specs
The following packages are incompatible
├─ gz-sim8 8.1.0  is installable with the potential options
│  ├─ gz-sim8 8.1.0 would require
│  │  └─ libgz-sim8 8.1.0 h7b813cf_0, which requires
│  │     └─ libabseil >=20230802.1,<20230803.0a0 , which can be installed;
│  ├─ gz-sim8 8.1.0 would require
│  │  ├─ gz-sim8-python >=8.1.0,<8.1.1.0a0  with the potential options
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libprotobuf >=4.25.2,<4.25.3.0a0 , which can be installed;
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 hb9b0dd2_1, which can be installed;
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 ha841f55_0, which can be installed;
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 h6914e37_2, which can be installed;
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 h7db9ec7_1, which can be installed;
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 hec09f77_0, which can be installed;
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 h59bcf31_2, which can be installed;
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 h86fce40_0, which does not exist (perhaps a missing channel);
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 hea5eaeb_2, which requires
│  │  │  │     ├─ libgz-common5 >=5.5.0,<6.0a0  with the potential options
│  │  │  │     │  ├─ libgz-common5 [5.5.0|5.5.1] would require
│  │  │  │     │  │  └─ libgdal [>=3.8.3,<3.9.0a0 |>=3.8.4,<3.9.0a0 ] with the potential options
│  │  │  │     │  │     ├─ libgdal [3.8.3|3.8.4] would require
│  │  │  │     │  │     │  └─ pcre2 >=10.42,<10.43.0a0 , which can be installed;
│  │  │  │     │  │     ├─ libgdal 3.8.4 would require
│  │  │  │     │  │     │  └─ libexpat >=2.6.1,<3.0a0 , which can be installed;
│  │  │  │     │  │     └─ libgdal [3.8.4|3.8.5|3.9.0] would require
│  │  │  │     │  │        └─ libexpat >=2.6.2,<3.0a0 , which can be installed;
│  │  │  │     │  └─ libgz-common5 5.6.0 would require
│  │  │  │     │     └─ libgdal [>=3.8.5,<3.9.0a0 |>=3.9.0,<3.10.0a0 ], which can be installed (as previously explained);
│  │  │  │     └─ python_abi 3.12.* *_cp312, which can be installed;
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 h1543f49_1, which can be installed;
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 haf7c398_2, which can be installed;
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 ha4001e2_0, which can be installed;
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 hffa47d9_2, which can be installed;
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 h7b813cf_0, which can be installed (as previously explained);
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 hc9e7752_1, which can be installed;
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 h5ee49ea_2, which can be installed;
│  │  │  ├─ gz-sim8-python 8.1.0 would require
│  │  │  │  └─ libgz-sim8 8.1.0 h4fe8855_0, which can be installed;
│  │  │  └─ gz-sim8-python 8.1.0 would require
│  │  │     └─ libgz-sim8 8.1.0 ha27685f_1, which can be installed;
│  │  └─ libgz-sim8 8.1.0 h45c09bf_1, which requires
│  │     └─ python_abi 3.12.* *_cp312, which can be installed;
│  └─ gz-sim8 8.1.0 would require
│     └─ libgz-sim8 8.1.0 hea5eaeb_2, which can be installed (as previously explained);
├─ opencv 4.10.0  is not installable because there are no viable options
│  ├─ opencv 4.10.0 would require
│  │  ├─ libprotobuf >=4.25.3,<4.25.4.0a0 , which requires
│  │  │  └─ libabseil >=20240116.1,<20240117.0a0 , which conflicts with any installable versions previously reported;
│  │  └─ python_abi 3.10.* *_cp310, which conflicts with any installable versions previously reported;
│  ├─ opencv 4.10.0 would require
│  │  ├─ libprotobuf >=4.25.3,<4.25.4.0a0 , which cannot be installed (as previously explained);
│  │  └─ python_abi 3.11.* *_cp311, which conflicts with any installable versions previously reported;
│  ├─ opencv 4.10.0 would require
│  │  ├─ libopencv [4.10.0 qt5_py312hc811b9e_500|4.10.0 qt6_py312h9dd261f_600], which requires
│  │  │  ├─ libglib >=2.80.2,<3.0a0 , which requires
│  │  │  │  └─ pcre2 >=10.43,<10.44.0a0 , which conflicts with any installable versions previously reported;
│  │  │  └─ qt6-main >=6.7.1,<6.8.0a0 , which requires
│  │  │     └─ pcre2 >=10.43,<10.44.0a0 , which conflicts with any installable versions previously reported;
│  │  └─ libprotobuf >=4.25.3,<4.25.4.0a0 , which cannot be installed (as previously explained);
│  ├─ opencv 4.10.0 would require
│  │  ├─ libprotobuf >=4.25.3,<4.25.4.0a0 , which cannot be installed (as previously explained);
│  │  └─ python_abi 3.8.* *_cp38, which conflicts with any installable versions previously reported;
│  ├─ opencv 4.10.0 would require
│  │  ├─ libprotobuf >=4.25.3,<4.25.4.0a0 , which cannot be installed (as previously explained);
│  │  └─ python_abi 3.9 *_pypy39_pp73, which conflicts with any installable versions previously reported;
│  └─ opencv 4.10.0 would require
│     ├─ libprotobuf >=4.25.3,<4.25.4.0a0 , which cannot be installed (as previously explained);
│     └─ python_abi 3.9.* *_cp39, which conflicts with any installable versions previously reported;
└─ vtk 9.2.6  is not installable because there are no viable options
   ├─ vtk 9.2.6 would require
   │  ├─ python_abi 3.10.* *_cp310, which conflicts with any installable versions previously reported;
   │  └─ vtk-base [9.2.6 qt_py310h1234567_203|9.2.6 qt_py310h1234567_204|...|9.2.6 qt_py310h1234567_223], which requires
   │     └─ python_abi 3.10.* *_cp310, which conflicts with any installable versions previously reported;
   ├─ vtk 9.2.6 would require
   │  ├─ python_abi 3.11.* *_cp311, which conflicts with any installable versions previously reported;
   │  └─ vtk-base [9.2.6 qt_py311h1234567_203|9.2.6 qt_py311h1234567_204|...|9.2.6 qt_py311h1234567_223], which requires
   │     └─ python_abi 3.11.* *_cp311, which conflicts with any installable versions previously reported;
   ├─ vtk 9.2.6 would require
   │  └─ vtk-base [9.2.6 qt_py312h1234567_218|9.2.6 qt_py312h1234567_219|...|9.2.6 qt_py312h1234567_223], which requires
   │     └─ libexpat [<2.6 |>=2.5.0,<2.6.0a0 ], which conflicts with any installable versions previously reported;
   ├─ vtk 9.2.6 would require
   │  ├─ python_abi 3.8.* *_cp38, which conflicts with any installable versions previously reported;
   │  └─ vtk-base [9.2.6 qt_py38h1234567_203|9.2.6 qt_py38h1234567_204|...|9.2.6 qt_py38h1234567_223], which requires
   │     └─ python_abi 3.8.* *_cp38, which conflicts with any installable versions previously reported;
   └─ vtk 9.2.6 would require
      ├─ python_abi 3.9.* *_cp39, which conflicts with any installable versions previously reported;
      └─ vtk-base [9.2.6 qt_py39h1234567_203|9.2.6 qt_py39h1234567_204|...|9.2.6 qt_py39h1234567_223], which requires
         └─ python_abi 3.9.* *_cp39, which conflicts with any installable versions previously reported.
~~~

### Installed packages

```shell
.
```

### Environment info

```shell
.
```

**Comment by traversaro** [comment]
This is also currently blocking the `libopencv410` migration for `bipedal-locomotion-framework`, see https://conda-forge.org/status/migration/libopencv4100 and https://github.com/regro/cf-scripts/actions/runs/9487143888 .

**Comment by minrk** [comment]
After #328, this should be resolved once migration starts for vtk 9.3

**Comment by traversaro** [comment]
Fixed by https://github.com/conda-forge/vtk-feedstock/pull/328 (and for the specific case https://github.com/conda-forge/pcl-feedstock/pull/65).
